t/issues/t4906-job-list-alloc-bypass.sh

```diff
@@ -0,0 +1,25 @@
+#!/bin/bash -e
```

Comment:
Since you have the follow on commit which adds a regression test to the job-list module, maybe this separate regression test isn't required? The reason I ask is that the t4000-issues test runs all the reproducers serially and at this point is really slow, so avoiding another `flux start` test therein would keep the testsuite a little leaner.

Reply:
actually the follow on test doesn't test it perfectly, as the regression that tests the event `alloc {"bypass":1}`, whereas the new test added `alloc {"annotations}:<something>, "etc": 1}`.

But I could tweak it accordingly.

Reply:
Ah, that might be good just to save some unnecessary work for the issues test driver.

Reply:
ahhh this was a good idea, ended up there was a corner case with exception notes (which are optional).